hotshot-state-prover/src/service.rs

```diff
@@ -356,42 +366,34 @@ mod test {
         providers::Middleware,
         utils::{Anvil, AnvilInstance},
     };
-    use hotshot_contract_adapter::jellyfish::field_to_u256;
     use hotshot_stake_table::vec_based::StakeTable;
     use hotshot_types::light_client::StateSignKey;
     use jf_primitives::signatures::{SchnorrSignatureScheme, SignatureScheme};
     use jf_utils::test_rng;
     use std::process::Command;
 
     const STAKE_TABLE_CAPACITY_FOR_TEST: usize = 10;
+    const BLOCKS_PER_EPOCH: u32 = 10;
+    const NUM_INIT_VALIDATORS: u32 = (STAKE_TABLE_CAPACITY_FOR_TEST / 2) as u32;
+    const TEST_MNEMONIC: &str = "test test test test test test test test test test test junk";
 
     /// Init a meaningful ledger state that prover can generate future valid proof.
     /// this is used for testing purposes, contract deployed to test proof verification should also be initialized with this genesis
     ///
-    /// NOTE: please update `contracts/script/LightClientTest.s.sol`'s genesis with the stderr print info
     #[allow(clippy::type_complexity)]
     fn init_ledger_for_test() -> (
         ParsedLightClientState,
         Vec<BLSPubKey>,
         Vec<(StateSignKey, StateVerKey)>,
         StakeTable<BLSPubKey, StateVerKey, CircuitField>,
     ) {
-        let mut rng = test_rng();
-        let (qc_keys, state_keys) = key_pairs_for_testing(STAKE_TABLE_CAPACITY_FOR_TEST, &mut rng);
-        let st = stake_table_for_testing(STAKE_TABLE_CAPACITY_FOR_TEST, &qc_keys, &state_keys);
-        let threshold = st.total_stake(SnapshotVersion::LastEpochStart).unwrap() * 2 / 3;
-
-        let stake_table_comm = st.commitment(SnapshotVersion::LastEpochStart).unwrap();
-        let genesis = ParsedLightClientState {
-            view_num: 0,
-            block_height: 0,
-            block_comm_root: U256::from(42), // arbitrary value
-            fee_ledger_comm: U256::from(42), // arbitrary value
-            bls_key_comm: field_to_u256(stake_table_comm.0),
-            schnorr_key_comm: field_to_u256(stake_table_comm.1),
-            amount_comm: field_to_u256(stake_table_comm.2),
-            threshold,
-        };
+        let pp = MockSystemParam::init(BLOCKS_PER_EPOCH);
+        let ledger = MockLedger::init(pp, NUM_INIT_VALIDATORS as usize);
+
+        let genesis = ledger.get_state();
+        let qc_keys = ledger.qc_keys;
+        let state_keys = ledger.state_keys;
+        let st = ledger.st;
```

Comment:
No need to change anything but I wonder why we often use names that are extremely short (`F`, `st`, `pp`, `pk`, `pi`). I understand that if you want to do many math operations the longer names make the expressions hard to read and long but I feel like in this repo we don't need to be so brief everywhere and it makes reading the code a bit more straightforward.